eth/watchers/eventdecoder_test.go

```diff
@@ -0,0 +1,74 @@
+package watchers
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/livepeer/go-livepeer/eth/contracts"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEventDecoder_FindEventType(t *testing.T) {
+	dec, err := NewEventDecoder(stubBondingManagerAddr, contracts.BondingManagerABI)
+	require.Nil(t, err)
+
+	assert := assert.New(t)
+
+	// Test unknown contract address
+	log := newStubBaseLog()
+	log.Address = common.BytesToAddress([]byte("foo"))
+
+	_, err = dec.FindEventName(log)
+	assert.EqualError(err, "log not from known contract")
+
+	// Check that bondingManagerDecoder is used
+	log = newStubUnbondLog()
+
+	eventName, err := dec.FindEventName(log)
+	assert.Nil(err)
+	assert.Equal("Unbond", eventName)
+}
+
+func TestEventDecoder_Decode(t *testing.T) {
+	dec, err := NewEventDecoder(stubBondingManagerAddr, contracts.BondingManagerABI)
+	require.Nil(t, err)
+
+	assert := assert.New(t)
+
+	// Test unknown contract address
+	log := newStubBaseLog()
+	log.Address = common.BytesToAddress([]byte("foo"))
+
+	var dummyEvent struct {
+		foo string
+	}
+	err = dec.Decode("foo", log, dummyEvent)
+	assert.EqualError(err, "log not from known contract")
+
+	// Check that bondingManagerDecoder is used
+	log = newStubUnbondLog()
+
+	var unbondEvent contracts.BondingManagerUnbond
+	err = dec.Decode("Unbond", log, &unbondEvent)
+	assert.Nil(err)
+}
+
+func TestBondingManagerDecoder_FindEventType(t *testing.T) {
```

Comment:
If we in-line the code block in `findEventType()` into `FindEventType()` then we can move these tests into `TestEventDecoder_FindEventType`

Reply:
wdyt about having a seperate test like this for every contract ABI that we use with the event decoder or would that be overkill? 
This would mean renaming it to something like `TestEventDecoder_BondingManager_FindEventType`

Reply:
I'd be ok with not testing for every contract ABI right now because `FindEventType()` just does a mapping lookup (which has no direct dependence on the contract ABI used) and we're already checking that behavior (for known vs. unknown events). What might be worth testing is the logic that populates `e.topicToEventName` in `NewEventDecoder()` - the constructor should populate the mapping with *all* the events included in a contract ABI. A small test that might be worth adding is:

1. Create a custom ABI string with a few events (i.e. 2 or 3)
2. Pass the custom ABI string into `NewEventDecoder()`
3. Check if the values for `e.topicToEventName` are correct for the custom ABI string

Reply:
added in 7ce08e7